Cubical/Data/Int/Properties.agda

```diff
@@ -233,6 +205,55 @@ ind-assoc _·_ f g p q base m n (suc o) =
     where
     IH = ind-assoc _·_ f g p q base m n o
 
++-comm : ∀ m n → m + n ≡ n + m
++-comm m (pos n) = ind-comm _+_ pos sucInt refl sucInt+ +sucInt pos0+ m n
++-comm m (negsuc n) = ind-comm _+_ negsuc predInt refl predInt+ +predInt negsuc0+ m n
+
 +-assoc : ∀ m n o → m + (n + o) ≡ (m + n) + o
 +-assoc m n (pos o) = ind-assoc _+_ pos sucInt +sucInt refl (λ _ _ → refl) m n o
 +-assoc m n (negsuc o) = ind-assoc _+_ negsuc predInt +predInt refl +predInt m n o
+
+-- Alternate definition: Compose sucPathInt with itself n times. Transporting along this
+-- will be addition, transporting with it backwards will be subtraction.
+
+addEq subEq : ℕ → Int ≡ Int
+addEq zero = refl
+addEq (suc n) = compPath (addEq n) sucPathInt
+
+subEq n i = addEq n (~ i)
+
+_+'_ : Int → Int → Int
+m +' pos n    = transport (addEq n) m
+m +' negsuc n = transport (subEq (suc n)) m
+
+private
+  FamilyOfEquiv : (f : Int → Int → Int) → Set
+  FamilyOfEquiv f = (m : Int) → isEquiv (λ n → f n m)
+
+  L0 : ∀ n z → +negsuc (suc n) z ≡ +negsuc n (predInt z)
+  L0 0 z = refl
+  L0 (suc n) z = cong predInt (L0 n z)
+  
+  Lnegsuc : ∀ n z → z +' negsuc n ≡ +negsuc n z
+  Lnegsuc 0 z = refl
+  Lnegsuc (suc n) z = compPath (Lnegsuc n (predInt z)) (sym (L0 n z))
+
+  Lpos : ∀ n z → z +' pos n ≡ +pos n z
+  Lpos zero z = refl
+  Lpos (suc n) z = cong sucInt (Lpos n z)
+  
+  Lemma : ∀ m n → m +' n ≡ m + n
+  Lemma m (pos n) = Lpos n m
+  Lemma m (negsuc n) = Lnegsuc n m
+
++'≡+ : _+'_ ≡ _+_
++'≡+ i m n = Lemma m n i
+
+-- We directly get that addition by a fixed number is an equivalence
+-- without having to do any induction!
+isEquivAddInt' : (m : Int) → isEquiv (λ n → n +' m)
+isEquivAddInt' (pos n)    = isEquivTransport (addEq n)
+isEquivAddInt' (negsuc n) = isEquivTransport (subEq (suc n))
+
+isEquivAddInt : (m : Int) → isEquiv (λ n → n + m)
+isEquivAddInt = subst FamilyOfEquiv +'≡+ isEquivAddInt'
```

Comment:
Very nice! I think this is an excellent example of how using univalence can make programming and proving more convenient. For comparison it would be nice to have the "naive" proof that `_ + m` is an equivalence (by defining `_ - m` and proving that this is an inverse to `_ + m` and then using `isoToEquiv`). Could you do this as well?